sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java

```diff
@@ -0,0 +1,147 @@
+    ARIA_PROPERTIES.put("aria-labelledby", new AriaProperty("aria-labelledby", AriaPropertyType.IDLIST));
+    ARIA_PROPERTIES.put("aria-level", new AriaProperty("aria-level", AriaPropertyType.INTEGER));
+    ARIA_PROPERTIES.put("aria-live", new AriaProperty("aria-live", AriaPropertyType.TOKEN, "off", "assertive", "polite"));
+    ARIA_PROPERTIES.put("aria-modal", new AriaProperty("aria-modal", AriaPropertyType.BOOLEAN));
+    ARIA_PROPERTIES.put("aria-multiline", new AriaProperty("aria-multiline", AriaPropertyType.BOOLEAN));
+    ARIA_PROPERTIES.put("aria-multiselectable", new AriaProperty("aria-multiselectable", AriaPropertyType.BOOLEAN));
+    ARIA_PROPERTIES.put("aria-orientation", new AriaProperty("aria-orientation", AriaPropertyType.TOKEN, "horizontal", "vertical", "undefined"));
+    ARIA_PROPERTIES.put("aria-owns", new AriaProperty("aria-owns", AriaPropertyType.IDLIST));
+    ARIA_PROPERTIES.put("aria-placeholder", new AriaProperty("aria-placeholder", AriaPropertyType.STRING));
+    ARIA_PROPERTIES.put("aria-posinset", new AriaProperty("aria-posinset", AriaPropertyType.INTEGER));
+    ARIA_PROPERTIES.put("aria-pressed", new AriaProperty("aria-pressed", AriaPropertyType.TRISTATE));
+    ARIA_PROPERTIES.put("aria-readonly", new AriaProperty("aria-readonly", AriaPropertyType.BOOLEAN));
+    ARIA_PROPERTIES.put("aria-relevant", new AriaProperty("aria-relevant", AriaPropertyType.TOKENLIST, "additions", "removals", "text", "all"));
+    ARIA_PROPERTIES.put("aria-required", new AriaProperty("aria-required", AriaPropertyType.BOOLEAN));
+    ARIA_PROPERTIES.put("aria-roledescription", new AriaProperty("aria-roledescription", AriaPropertyType.STRING));
+    ARIA_PROPERTIES.put("aria-rowcount", new AriaProperty("aria-rowcount", AriaPropertyType.INTEGER));
+    ARIA_PROPERTIES.put("aria-rowindex", new AriaProperty("aria-rowindex", AriaPropertyType.INTEGER));
+    ARIA_PROPERTIES.put("aria-rowspan", new AriaProperty("aria-rowspan", AriaPropertyType.INTEGER));
+    ARIA_PROPERTIES.put("aria-selected", new AriaProperty("aria-selected", AriaPropertyType.BOOLEAN, true, "true", "false"));
+    ARIA_PROPERTIES.put("aria-setsize", new AriaProperty("aria-setsize", AriaPropertyType.INTEGER));
+    ARIA_PROPERTIES.put("aria-sort", new AriaProperty("aria-sort", AriaPropertyType.TOKEN, "ascending", "descending", "none", "other"));
+    ARIA_PROPERTIES.put("aria-valuemax", new AriaProperty("aria-valuemax", AriaPropertyType.NUMBER));
+    ARIA_PROPERTIES.put("aria-valuemin", new AriaProperty("aria-valuemin", AriaPropertyType.NUMBER));
+    ARIA_PROPERTIES.put("aria-valuenow", new AriaProperty("aria-valuenow", AriaPropertyType.NUMBER));
+    ARIA_PROPERTIES.put("aria-valuetext", new AriaProperty("aria-valuetext", AriaPropertyType.STRING));
+  }
+
+  static class AriaProperty {
+    private final String name;
+    private final AriaPropertyType type;
+    private final Optional<Boolean> allowUndefined;
+    private final List<String> values;
```

Comment:
why not use a Set?

Reply:
Sure, but I won't be able to assert some issue messages based on this set as the value order won't be deterministic on collection.

Reply:
You did the change, meaning it's all good?

Reply:
Yes